### ISSUE TITLE
NER model is not predicting any labels, even on test set during traning

### ISSUE THREAD

**Issue by LyaSolis** [smitkiri/ehr-relation-extraction#20]
Hi, thank you for this repo, unfortunately I ran into some issues and because there is no error message, I am unable to debug it....

When I try to generate predictions, I get blank output (model isn't predicting anything).
I realised that during training it also failed to predict any entities, but there was no error message, just warnings.

I used the following script to prepare dataset:
```
!python generate_data.py --task ner   \
                  --input_dir ./data/   \
                  --ade_dir ./ade_corpus/  \
                   --target_dir biobert_ner/dataset/   \
                  --max_seq_len 512 --dev_split 0.1   \
                  --tokenizer biobert-base   \
                   --ext txt --sep " "  \
```
Log:

```
Reading data

Train data:
Progress: [====================] 303/303

Test Data:
Progress: [==================> ] 191/202/Users/lsolis/Documents/GitHub/PubMed_pipeline/ehr-relation-extraction/ehr.py:186: UserWarning: Invalid annotation encountered: ['12 hours'], File: ./data/test/106967.ann
  warnings.warn(msg)
Progress: [====================] 202/202

ADE data: Done


Data successfully saved in biobert_ner/dataset/train.txt
Variable successfully saved in biobert_ner/dataset/train.pkl
Data successfully saved in biobert_ner/dataset/train_dev.txt
Variable successfully saved in biobert_ner/dataset/train_dev.pkl
Data successfully saved in biobert_ner/dataset/devel.txt
Variable successfully saved in biobert_ner/dataset/devel.pkl
Data successfully saved in biobert_ner/dataset/test.txt
Variable successfully saved in biobert_ner/dataset/test.pkl

Generating files successful. Files generated: train.txt, train.pkl, train_dev.txt, train_dev.pkl, devel.txt, devel.pkl, test.txt, test.pkl, labels.txt
```
Then `run_ner.py` 
```
cd biobert_ner
export SAVE_DIR=./output1
export DATA_DIR=./dataset

export MAX_LENGTH=128
export BATCH_SIZE=16
export NUM_EPOCHS=5
export SAVE_STEPS=1000
export SEED=0

python run_ner.py --data_dir ${DATA_DIR}/ --labels ${DATA_DIR}/labels.txt --model_name_or_path dmis-lab/biobert-large-cased-v1.1 --output_dir ${SAVE_DIR}/ --max_seq_length ${MAX_LENGTH} --num_train_epochs ${NUM_EPOCHS} --per_device_train_batch_size ${BATCH_SIZE} --save_steps ${SAVE_STEPS} --seed ${SEED} --do_train --do_eval --do_predict --overwrite_output_dir
```

At the very last stage there was a print out in terminal, here is a fraction of it
```
05/10/2022 04:38:49 - WARNING - __main__ -   Example 1966, Example: 8 O

05/10/2022 04:38:49 - WARNING - __main__ -   Example 1966, Example: . O

05/10/2022 04:38:49 - WARNING - __main__ -   Example 1966, Example: per B-DRUG

05/10/2022 04:38:49 - WARNING - __main__ -   Example 1966, Example: 5 B-STR
```  

`test_results.txt` content
```
eval_loss = 1.8185148617116416
eval_precision = 0.0
eval_recall = 0.0
eval_f1 = 0.0
```

And the fraction of `test_predictions.txt` - they all are "O", there is not a single other label predicted...
```
##9 O
##am O
##b O
##c O
##b O
##c O
##g O
##b O
##ct O
##c O
##v O
##ch O
##ch O
##c O
##d O
##w O
##lt O
##t O
##9 O
##am O
##uts O
##ymph O
##s O
##os O
##os O
##as O
##o O
##y O
```

**Comment by smitkiri** [comment]
There was a similar issue created a while ago [here](https://github.com/smitkiri/ehr-relation-extraction/issues/1). Not sure how they solved it, but there was an issue with GCP. Where are you training the model?

**Comment by LyaSolis** [comment]
Thank you for your reply! 

It seems that NER and RE were implemented in different Python versions. My guess is - Google Colab upgraded their default version when you were transitioning from one task to the other?

NER task requires Python3.7 
RE - Python3.8

Running RE in Python3.7 gives the same error as when I run NER in Python3.8! 

Maybe it's worth updating README - because it took me quite some time to figure it out - requirements get installed in both Python versions with no issues at all, so there is nothing to raise a flag regarding versioning.

Other than that - amazing work, thank you! :)

**Comment by smitkiri** [comment]
Interesting. I used the same python version to train both NER and RE tasks (Python 3.8.3). I didn't use Google Colab for training, so I'm not sure if this is a google colab specific issue. But feel free to add a troubleshooting section in the README and pull up a PR!
